### PR TITLE
Add optional global nav class-driven font iconography

### PR DESCRIPTION
**Pull request by ghost**
Commit name should say it all. I chose `class` over `icon` for extensibility and continuity with the footer. Sorry I don't have time to add documentation.

screen shot 2016-12-01 at 3 29 32 pm

screen shot 2016-12-01 at 3 43 14 pm

screen shot 2016-12-01 at 3 44 19 pm

## Example use from `navigation.yml`

```yml
- title: Start
  url: "/"
  class: "icon-home"
  side: left
```

## Included with pull

- Replaces the `tree` font icon with `xing` for hacker appeal
- Also replaces hamburger helper text "Navigation" with "Nav" to support longer site titles

### CODE REVIEW

File: _sass/_07_layout.scss [hunk 1]
```diff
@@ -48,6 +48,10 @@ body.video cite { color: #fff; }
 #navigation {
     -webkit-box-shadow: 0 2px 2px 0 rgba(0,0,0,.2);
     box-shadow: 0 2px 3px 0 rgba(0,0,0,.2);
+
+    [class^='icon-']:before, [class*=' icon-']:before {
```
Comment (ghost) [inline _sass/_07_layout.scss:52]: Thanks to @TWiStErRob for this one. Modified [from `twisterrob.github.io`](https://github.com/TWiStErRob/twisterrob.github.io/blob/0ba9f2696ba3ec1105166927f0e1bdd4e3dfb2bd/_sass/_07_layout.scss#L60-L63).

**Comment by ghost** [comment]
FYI tested this on Android and iOS physical devices in addition to emulator. Real device testing reveals a possible bug on iOS devices. Adding screenshots for each case...

**Comment by ghost** [comment]
Mobile emulator for iPhone 6 in Chrome Dev Tools:

screen shot 2016-12-01 at 3 43 14 pm

Actual experience on iPhone 6 (not 6 Plus) without adjustment to Settings > Accessibility:

screen shot 2016-12-02 at 8 57 06 am


Android experience. Device and settings unknown. Sent from client.

screen shot 2016-12-02 at 8 57 25 am

Not sure what's causing the clipping, but didn't see this bug prior to these changes.

**Comment by Phlow** [comment]
Looks good. But why did you choose the Xing-icon? This is an icon for the biggest German social network for business.

**Comment by jhabdas** [comment]
Hah. No kidding?  Please tell me, what is Germany's business site/app? Very curious to see.

**Comment by jhabdas** [comment]
Oh. I see it now. xing.com. thanks for the info.

File: _includes/_navigation.html [hunk 2]
```diff
@@ -2,7 +2,7 @@
   <nav class="top-bar" role="navigation" data-topbar>
     <ul class="title-area">
       <li class="name">
-      <h1 class="show-for-small-only"><a href="{{ site.url }}{{ site.baseurl }}" class="icon-xing"> {{ site.title }}</a></h1>
+      <h1 class="show-for-small-only"><a href="{{ site.url }}{{ site.baseurl }}" class="icon-tree"> {{ site.title }}</a></h1>
```
Comment (ghost) [inline _includes/_navigation.html:5]: LGTM. Thanks for the heads-up on the German social sharing site.

**Comment by Phlow** [comment]
Yeah, it's Xing.com – it is the big German counterpart of LinkedIn.

**Pull request merged**